src/apps/media/src/app/components/MediaDetailsModal.js

```diff
@@ -89,7 +88,8 @@ export const MediaDetailsModal = memo(function MediaDetailsModal(props) {
               </label>
             }
             placeholder={"Image Filename"}
-            onChange={(val) => setFilename(val)}
+            // Replaces all non-alphanumeric characters (excluding '.') with '-' to reflect the filename transformation done on the BE
+            onChange={(val) => setFilename(val.replaceAll(/[^a-z\d-.]/gi, "-"))}
```

Comment:
there is a format util function, we use it when adding field labels, which ensures computer friendly strings. we need to be using that here.